tap_exacttarget/endpoints/data_extensions.py

```diff
@@ -120,13 +140,61 @@ def _get_fields(self, extensions):
                 [extension_id, 'schema', 'properties', field_name],
                 field_schema)
 
+            if is_primary_key:
+                for mdata in to_return[extension_id]['metadata']:
+                    if not mdata.get('breadcrumb'):
+                        mdata.get('metadata').get('table-key-properties').append(field_name)
+
+            if is_replication_key:
+                for mdata in to_return[extension_id]['metadata']:
+                    if not mdata.get('breadcrumb'):
+                        mdata.get('metadata')['forced-replication-method'] = "INCREMENTAL"
+                        mdata.get('metadata').get('valid-replication-keys').append(field_name)
+
             # These fields are defaulted into the schema, do not add to metadata again.
             if field_name not in {'_CustomObjectKey', 'CategoryID'}:
-                to_return[extension_id]['metadata'].append({
-                    'breadcrumb': ('properties', field_name),
-                    'metadata': {'inclusion': 'available'}
-                })
-
+                if is_primary_key or is_replication_key:
+                    to_return[extension_id]['metadata'].append({
+                        'breadcrumb': ('properties', field_name),
+                        'metadata': {'inclusion': 'automatic'}
+                    })
+                else:
+                    to_return[extension_id]['metadata'].append({
+                        'breadcrumb': ('properties', field_name),
+                        'metadata': {'inclusion': 'available'}
+                    })
+
+        # to_return =
```

Comment:
@harshpatel4crest What is the purpose of this commented code?

Reply:
Updated the code with more information for reference